### PR TITLE
Add initial iMX8 support

### PR DESCRIPTION
**Pull request by lrusak**
This add support for the iMX8 platform by NXP.

This is just the initial support for the platform. Unfortunately the hdmi controller hasn't hit mainline yet and I plan to add it as a patch in a later PR. This PR will get LibreELEC booting to a basic instance.

I've had some issues with mainline u-boot on the iMX8 boards I have but hopefully this will be worked out in the future. I currently have to manually adjust the vendor installed u-boot to boot LibreELEC.

iMX8 has a hantro video decoder which uses the v4l2 stateless patches.

It looks to me like we have a mixture of options to advertise V4L2_SUPPORT in `config/graphics` and in the `ffmpeg/package.mk`. This should probably be improved in the future.

### CODE REVIEW

File: projects/NXP/devices/iMX8/bootloader/firmware [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2017-present Team LibreELEC (https://libreelec.tv)
+
+[ -n "$ATF_PLATFORM" ] && cp -av $(get_install_dir atf)/usr/share/bootloader/bl31.bin .
+
+cp -av $(get_build_dir firmware-imx)/firmware/hdmi/cadence/signed_hdmi_imx8m.bin $(get_build_dir $BOOTLOADER)/
```
Comment (jernejsk) [inline projects/NXP/devices/iMX8/bootloader/firmware:7]: Are you sure this is needed? AFAIK Blobs for HDMI are usually needed only for HDCP 2.2.

Comment (lrusak) [reply projects/NXP/devices/iMX8/bootloader/firmware:7]: Yes it's needed
```
WARNING 'signed_hdmi_imx8m.bin' not found, resulting binary is not-functional
  CFGS    spl/u-boot-spl.cfgout
  MKIMAGE flash.bin
./tools/mkimage: Can't open spl/u-boot-spl-ddr.bin: No such file or directory
make[1]: *** [arch/arm/mach-imx/Makefile:158: flash.bin] Error 1
```

**Comment by lrusak** [comment]
objections?

**Comment by jernejsk** [comment]
No, seems clean to me.

**Review by jernejsk — APPROVED** [review]

**Comment by CvH** [comment]
@lrusak is somewhere documented how these images are build ? 
Otherwise pls add them here https://github.com/LibreELEC/documentation/blob/master/development-1/build-commands/build-commands-le10.md

**Comment by heitbaum** [comment]
> @lrusak is somewhere documented how these images are build ?
> Otherwise pls add them here https://github.com/LibreELEC/documentation/blob/master/development-1/build-commands/build-commands-le10.md

https://github.com/LibreELEC/documentation/pull/14

**Pull request merged**